datasources/twitter/api_test.go

```diff
@@ -0,0 +1,308 @@
+	assertEqualString(t, acc.ProfileSidebarBorderColor, "AAAAAA")
+	assertEqualString(t, acc.ProfileSidebarFillColor, "FFFFFF")
+	assertEqualString(t, acc.ProfileTextColor, "000000")
+
+	latestTweet := acc.Status // shorthand
+
+	assertEqualString(t, latestTweet.TweetIDStr, "1234567890123456789")
+	assertTrue(t, latestTweet.TweetID == 1234567890123456789)
+	assertTrue(t, latestTweet.User == nil)
+	assertEqualString(t, latestTweet.CreatedAt, "Wed Nov 27 18:54:49 +0000 2019")
+	assertEqualString(t, latestTweet.Text, "Demo tweet #testing https://t.co/abcdefgijk")
+	assertEqualString(t, latestTweet.FullText, "")
+	assertEqualString(t, latestTweet.Lang, "en")
+	assertEqualString(t, latestTweet.Source, `<a href="http://instagram.com" rel="nofollow">Instagram</a>`)
+	assertTrue(t, !latestTweet.Truncated)
+	assertTrue(t, !latestTweet.PossiblySensitive)
+	assertTrue(t, !latestTweet.IsQuoteStatus)
+
+	assertEqualString(t, latestTweet.InReplyToScreenName, "")
+	assertTrue(t, latestTweet.InReplyToStatusID == 0)
+	assertEqualString(t, latestTweet.InReplyToStatusIDStr, "")
+	assertTrue(t, latestTweet.InReplyToUserID == 0)
+	assertEqualString(t, latestTweet.InReplyToUserIDStr, "")
+
+	assertTrue(t, !latestTweet.WithheldCopyright)
+	assertTrue(t, len(latestTweet.WithheldInCountries) == 0)
+	assertEqualString(t, latestTweet.WithheldScope, "")
+
+	assertTrue(t, !latestTweet.Favorited)
+	assertTrue(t, latestTweet.FavoriteCount == 0)
+
+	assertTrue(t, !latestTweet.Retweeted)
+	assertTrue(t, latestTweet.RetweetedStatus == nil)
+	assertTrue(t, latestTweet.RetweetCount == 0)
+
+	assertTrue(t, len(latestTweet.DisplayTextRange) == 0)
+
+	assertTrue(t, latestTweet.Coordinates.Latitude() == 34.0522)
+	assertTrue(t, latestTweet.Coordinates.Longitude() == -118.243)
+
+	assertTrue(t, latestTweet.ExtendedEntities == nil)
+	// I was too lazy to type assertions for the "entities" hierarchy, so we're just comparing
+	// re-serialized versions. this would catch if we would have had typos in JSON field
+	// names (they would not get decoded, and hence would not get re-serialized)
+	entitiesJson, err := json.MarshalIndent(latestTweet.Entities, "", "  ")
+	assertTrue(t, err == nil)
+	assertEqualString(t, string(entitiesJson), `{
+  "hashtags": [
+    {
+      "indices": [
+        0,
+        8
+      ],
+      "text": "testing"
+    }
+  ],
+  "symbols": [],
+  "user_mentions": [],
+  "urls": [
+    {
+      "url": "https://t.co/abcdefgijk",
+      "expanded_url": "https://www.instagram.com/p/BAABAABAABA/?igshid=xyxyxyxyxyxyx",
+      "display_url": "instagram.com/p/BAABAABAABA/…",
+      "indices": [
+        52,
+        75
+      ]
+    }
+  ],
+  "polls": null
+}`)
+}
+
+func assertEqualString(t *testing.T, actual string, expected string) {
+	t.Helper()
+
+	if actual != expected {
+		t.Fatalf("exp=%v; got=%v", expected, actual)
+	}
+}
+
+func assertTrue(t *testing.T, val bool) {
+	t.Helper()
+
+	if !val {
+		t.Fatal("expected true; got false")
+	}
+}
```

Comment:
I didn't know if it would be acceptable to add any dependencies like testing helpers, so I just copy pasted the stuff I use from https://github.com/function61/gokit/blob/master/assert/assert.go